RxSwift/Observables/Max.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  Max.swift
+//  RxSwift
+//
+//  Created by Shai Mishali on 8/19/17.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+extension ObservableType {
+
+    /**
+     Emits the element from the source Observable that had the maximum value using a provided comparator closure.
+
+     - parameter comparator: A comparator method expected to return true if the first element has a greater value then the second element.
+     - returns: An observable sequence containing the specified number of elements from the end of the source sequence.
+    */
+    public func max(_ comparator: @escaping (E, E) -> Bool)
```

Comment:
It seems like there is not difference between `Max` and `Min` producer or I missing something?

I feel like there could be only one implementation but with different comparison strategies.
E.g 
enum ComparisonStrategy {
    case min
    case max
}

So you just swap the order of comparison. Min: e2 < e1, Max: e1 < e2.

Am I right?